.github/workflows/pythonpip.yml

```diff
@@ -0,0 +1,45 @@
+name: Cache Python and pip dependencies
+# This workflow is triggered every 2 days and updates the Python
+# and pip dependencies cache
+
+on:
+  schedule:
+    - cron: '0 0 */2 * *'  # This triggers the workflow every 2 days at midnight
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.9', '3.10', '3.11', '3.12']
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Cache Python environment
+        id: cache-env
+        uses: actions/cache@v3
+        with:
+          path: ./.venv-${{ matrix.python-version }}
+          key: ${{ runner.os }}-venv-${{ matrix.python-version }}-${{ hashFiles('devtools/dev-requirements.txt') }}
```

Comment:
@f0uriest Ok so this line will hash the contents of the file, so any change to the file will create a new cache. But the problem is if you update `requirements.txt` I don't think it will be reflected to the hashing since `devtools/dev-requirements.txt` only has is as `-r requirements.txt`. Any suggestions?

Reply:
I can change it like `-${{ hashFiles('devtools/dev-requirements.txt') }}-${{ hashFiles('requirements.txt') }}`